### ISSUE TITLE
CrudRepository is postgres only

### ISSUE THREAD

**Issue by farfromrefug** [Seb-C/kiss-orm#19]
[here](https://github.com/Seb-C/kiss-orm/blob/master/src/Repositories/CrudRepository.ts#L123) you use `RETURNING` which for example wont work with sqlite.
Is `CrudRepository` meant to be postgres only?

**Comment by Seb-C** [comment]
Currently the queries are indeed only compatible with PostgreSQL, but I want to change this design, as I partially documented [here](https://github.com/Seb-C/kiss-orm/issues/3#issuecomment-693155119).

The update method can be replaced with an UPDATE + SELECT, but there is no standard way to handle the INSERT + SELECT.

So I want to add a method in the `Database` class that receives a standard `INSERT` request and executes it + returns the inserted auto-increment id (if any) or inserted data. From there, the repository class would have to either return the data, perform a select using this id, or a select using the explicitly inserted value for the primary key (uuids case for example).
It is not a very elegant design, but I cannot think of anything else that keeps the code simple while working with all databases.

**Comment by farfromrefug** [comment]
@Seb-C I like your idea . It makes sense to let the Database handle the "query"
I managed to have a working repository. Had to remove "hard core" the `RESULTING` reference but after that this :
```ts
import SqlQuery from 'kiss-orm/dist/Queries/SqlQuery';
import DatabaseInterface from 'kiss-orm/dist/Databases/DatabaseInterface';
import { SQLiteDatabase, openOrCreate } from '@akylas/nativescript-sqlite';
const sql = SqlQuery.createFromTemplateString;

export default class NSQLDatabase implements DatabaseInterface {
    db: SQLiteDatabase;
    constructor(
        filePath: string,
        options?: {
            threading?: boolean;
            flags?: number;
        }
    ) {
        this.db = openOrCreate(filePath, options);
    }
    indexToPlaceholder(i: number): string {
        return '?';
    }
    formatIdentifier(i: string): string {
        return i;
    }
    async disconnect() {
        await this.db.close();
    }
    async query(query: SqlQuery): Promise<any[]> {
        const compiledQuery = query.compile(this.indexToPlaceholder, this.formatIdentifier);
        const sqlQuery = compiledQuery.sql.trim();
        const isInsertQuery = sqlQuery.startsWith('INSERT INTO') || sqlQuery.startsWith('UPDATE');
        let result = isInsertQuery // when isInsertQuery == true, result is the id
            ? await this.db.execute(sqlQuery, compiledQuery.params)
            : await this.db.select(sqlQuery, compiledQuery.params);
        if (isInsertQuery && !result) {
            // create await an array result.
            result = [];
        }
        return result as any[];
    }

    sequence<T>(sequence: (query: (query: SqlQuery) => Promise<any[]>) => Promise<T>) {
        return this.db.transaction<T>(() => sequence((query: SqlQuery) => this.query(query) as any) as any) as any;
    }
}
```
It works well. Right now the only thing not done well is that the query method does not return anything(`db` side) so i have to do that trick where i return an Array.

**Comment by Seb-C** [comment]
Thanks for sharing. This is definitely an issue that I want to work on (as well as the other opened issues), but I don't know when I will have time to do so yet.

**Comment by Seb-C** [comment]
This will be fixed with v2.0.0, see #22

**Comment by Seb-C** [comment]
v2 has now been published, including MySQL and SQLite compatibility.
